contracts/utils/Multipass.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: MIT
+// Based on OpenZeppelin Contracts (utils/Multicall.sol)
+
+pragma solidity ^0.8.28;
+
+import { Context } from "@openzeppelin/utils/Context.sol";
+
+/**
+ * @dev Provides a function to batch together multiple calls in a single external call without reverting on failure.
+ *
+ * Consider any assumption about calldata validation performed by the sender may be violated if it's not especially
+ * careful about sending transactions invoking {multipass}. For example, a relay address that filters function
+ * selectors won't filter calls nested within a {multipass} operation.
```

Comment:
What happens when one call fails? Do the state changes of that specific call get reverted while the rest will continue to pass?

Reply:
That's my understanding.